src/libmongoc/src/mongoc/mongoc-openssl.c

```diff
@@ -613,32 +784,33 @@ _mongoc_ocsp_tlsext_status_cb (SSL *ssl, void *arg)
    }
 
    /* Get the stapled OCSP response returned by the server */
-   len = SSL_get_tlsext_status_ocsp_resp (ssl, &r);
-   stapled_response = !!r;
+   len = SSL_get_tlsext_status_ocsp_resp (ssl, &resp_data);
+   stapled_response = !!resp_data;
    if (stapled_response) {
       /* obtain an OCSP_RESPONSE object from the OCSP response */
-      if (!d2i_OCSP_RESPONSE (&resp, &r, len)) {
+      if (!d2i_OCSP_RESPONSE (&resp, &resp_data, len)) {
          MONGOC_ERROR ("Failed to parse OCSP response");
          ret = OCSP_CB_ERROR;
          GOTO (done);
       }
    } else {
-      MONGOC_DEBUG ("Server does not contain a stapled response");
-      must_staple = X509_get_ext_d2i (peer, NID_tlsfeature, 0, 0) != NULL;
```

Comment:
This previously checked for the existence of the tlsfeature extension. But after looking at the server code I realized this should be looking for the status_request extension within the tlsfeature extension:
https://github.com/mongodb/mongo/blob/4dcdc501520558d602897aed9aa61781265328d7/src/mongo/util/net/ssl_manager_openssl.cpp#L1605-L1616

pymongo does the same:
https://github.com/mongodb/mongo-python-driver/blob/29960237dcdd9d90402d6c76c4296aec3d01d3fc/pymongo/ocsp_support.py#L292-L298